File: src/test/java/com/raychenon/leetcode/LongestArithmeticSubsequenceofGivenDifferenceTest.java
package com.raychenon.leetcode;

import org.junit.BeforeClass;
import org.junit.Test;

import static junit.framework.TestCase.assertEquals;

/**
 * User: raychenon
 * Date: 2019-10-08
 * https://leetcode.com/problems/longest-arithmetic-subsequence-of-given-difference/
 */
public class LongestArithmeticSubsequenceofGivenDifferenceTest {


    private static LongestArithmeticSubsequenceofGivenDifference instance;

    @BeforeClass
    public static void setUp() {
        instance = new LongestArithmeticSubsequenceofGivenDifference();
    }

    @Test
    public void example1Test() {
        int input[] = {1, 2, 3, 4};
        assertEquals(4, instance.longestSubsequence(input, 1));
    }

    @Test
    public void example2Test() {
        int input[] = {1, 3, 5, 7};
        assertEquals(1, instance.longestSubsequence(input, 1));
    }

    @Test
    public void example3Test() {
        int input[] = {1, 5, 7, 8, 5, 3, 4, 2, 1};
        assertEquals(4, instance.longestSubsequence(input, -2));
    }
}

File: src/test/kotlin/com/raychenon/kotlin/leetcode/LongestArithmeticSubsequenceofGivenDifferenceTest.kt
package com.raychenon.kotlin.leetcode

import org.junit.Assert
import org.junit.Test

/**
 * User: raychenon
 * Date: 2019-10-08
 * https://leetcode.com/problems/longest-arithmetic-subsequence-of-given-difference/
 */
class LongestArithmeticSubsequenceofGivenDifferenceTest {

    @Test
    fun example1Test() {
        Assert.assertEquals(
            4,
            LongestArithmeticSubsequenceofGivenDifference.longestSubsequence(intArrayOf(1, 2, 3, 4), 1)
        )
    }

    @Test
    fun example2Test() {
        Assert.assertEquals(
            1,
            LongestArithmeticSubsequenceofGivenDifference.longestSubsequence(intArrayOf(1, 3, 5, 7), 1)
        )
    }

    @Test
    fun example3Test() {
        Assert.assertEquals(
            4,
            LongestArithmeticSubsequenceofGivenDifference.longestSubsequence(intArrayOf(1, 5, 7, 8, 5, 3, 4, 2, 1), -2)
        )
    }
}